plugins/nodes/src/js/data/MesosMasters.js

```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import { Observable } from "rxjs/Observable";
+import "rxjs/add/observable/timer";
+import "rxjs/add/operator/do";
+import "rxjs/add/operator/map";
+import "rxjs/add/operator/retry";
+import "rxjs/add/operator/filter";
+
+import MesosStateStore from "#SRC/js/stores/MesosStateStore";
+
+import LeaderGrid from "../components/LeaderGrid";
+
+function isEmptyObject(obj) {
+  return Object.keys(obj).length === 0;
+}
+
+function hostPort(address) {
+  return `${address.hostname}:${address.port}`;
+}
+
+function supportsRegion(master) {
+  return (
+    master.domain &&
+    master.domain.fault_domain &&
+    master.domain.fault_domain.region &&
+    master.domain.fault_domain.region.name
+  );
+}
+
+export function getRegion(master) {
+  return supportsRegion(master)
+    ? master.domain.fault_domain.region.name
+    : "N/A";
+}
+
+const STORE_POLL_INTERVAL = 2000;
+
+function mesosMasterEmpty() {
+  return {
+    hostPort: undefined,
```

Comment:
Why do you think, the backend returned value matters? I wanted to define an initial value for the component state. I think both null and undefined work the same for me. I just kept undefined because that is the way it was on the component I copied this logic from.

Reply:
That way we keep the expected "types".

Got it.